data_architect_misc/data_transformer/transform_functions/eu_norway_transform_functions.py

```diff
@@ -0,0 +1,60 @@
+"""This is the subclass of Transform function for Norway (EU division).
+
+We will define transform functions specific to Norway here.
+
+Author: Maicol Contreras
+Last Modified: December 21, 2020
+"""
+from constants import comp_harm_constants
+import pandas as pd
+import xlrd
+from transform_functions.common_comp_harm_transform_functions import CommonCompHarmTransformFunctions
+
+
+class EuNorwayTransformFunctions(CommonCompHarmTransformFunctions):
+    """
+    All custom (uncommon) transform functions **SPECIFIC to
+    individual processing task** must be defined as part
+    of this class.
+    """
+    NORWAY_SPECIFIC_CATEGORY_MAPPINGS = {
+        # Add any Norway-specific mappings that cannot be used for other EU countries here.
+    }
+
+    def __init__(self, config):
+        self.config = config
+        self.category_mappings = dict(
+            comp_harm_constants.ENGLISH_CATEGORY_MAPPINGS,
+            **EuNorwayTransformFunctions.NORWAY_SPECIFIC_CATEGORY_MAPPINGS)
+
+    @staticmethod
+    def _columns_reference(columns_ref):
+        # In France raw file, we need to validate that all columns always come with the same name
+        # In some ocassions, the local agency shares with us the data with other column but the same data.
+        # We make sure the name of the columns using a remap of columns.
+        try:
+            new_column = { 
+                "grossprice" : "gross"
+                }
+            column = new_column[columns_ref.lower()]
+        except:
+            column = columns_ref
+        finally:
+            return column
+
+    def join_sheets_in_a_unique_dataframe(self,df):
+        columns_to_use = ['category', 'subcategory', 'brand', 'product', 'advertiser', 'period', 'media channel', 'gross']
+        final_df = pd.DataFrame(columns=columns_to_use)
+        wb = xlrd.open_workbook(self.config['current_input_file'])
```

Comment:
Why do we use xlrd here? Please use pandas (pd.read_excel) because we don't want to rely on two different libraries to open an Excel/CSV file.

Reply:
Sire, this is because I validate the sheets that are actived and then we merge the sheets using every sheet name with Pandas